Held suarez docs (#189)
ExeClim/Isca: docs/source/modules/held_suarez.rst

Dry dynamical core (Held--Suarez) Forcing  
=======================================================================================

Documentation for ``hs_forcing_mod``. 

Summary
-------

> This module is contains the 'physics options' used by Isca when the flag ``idealised_moist_model`` in ``atmosphere_nml`` is set to ``FALSE``. 

Reviewer: Do we also want to make clear that if would be preferable to do`from isca import DryCodeBase` for this, not just setting the `idealised_moist_model` flag?

> This module is contains the 'physics options' used by Isca when the flag ``idealised_moist_model`` in ``atmosphere_nml`` is set to ``FALSE``. 

Reviewer: I think we need American spelling for this one...
```suggestion
This module is contains the 'physics options' used by Isca when the flag ``idealized_moist_model`` in ``atmosphere_nml`` is set to ``FALSE``. 
```

By default, :math:`T_{\text{strat}}=200\,\text{K}` (as above), :math:`p_{\text{trop}}=0.1\,\text{bar}` and :math:`\alpha=2/7`. These can all be changed in the namelist. 

| **Equilibrium temperature profile from file**
| When the ``'from_file'`` option is specified, then :math:`T_{\text{eq}}` is computed from a specified input file. The file should be a netCDF file, and temperature can be defined as a function of time, pressure, latitude, and longtiude (with time optional). 

The zonal mean of :math:`T_{\text{eq}}` provided in the file is taken, and this is used as the relaxation temperature field for the Newtonian damping. 

The name of the netCDF file is specified with the namelist option ``equilibrium_t_file``, and should exclude the ``.nc`` suffix. Note: in the netCDF file, the :math:`T_{\text{eq}}` variable must be the same name as the name of the file, i.e., ``equilibrium_t_file``. The path to the file should be configured in the model's python run script (see the test cases for examples of how to do this). 

> Top-down Newtonian damping 

Reviewer: As a note, I have found the stability of the top-down routine to be very dependent on the choice of sigma levels. I think I only managed to get it to converge using `even_sigma` (as it is in the test case). I'm not sure if this is more widely known, but perhaps it should be noted here if so?

The heating is maximal at the surface and decays with pressure. :math:`\lambda_{0}` and :math:`\theta_{0}` specify the central longitude and latitude of the heating, respectively, :math:`\Delta\lambda` and :math:`\Delta\theta` specify the width of the heating, and :math:`\Delta p` determines how quickly the heating decays away from the surface. :math:`A` is the amplitude of the forcing at :math:`p=p_{\text{s}}`, :math:`\lambda=\lambda_{0}`, :math:`\theta=\theta_{0}`. Each of these parameters may be specified in the namelist. 

| **From file** 
| When ``local_heating_option = 'from_file'``, :math:`Q_{\text{local}}` is obtained from an input netCDF file (with possible dimensions: time, pressure, latitude, longitude, where time is optional). 

The name of the netCDF file is specified with the namelist option ``local_heating_file``, and should exclude the ``.nc`` suffix. Note: in the netCDF file, the ``tdt`` temperature tendency variable must be the same name as the file name, i.e., ``local_heating_file``. The path to the file should be configured in the model's python run script (see the test cases for examples of how to do this). 

   

Namelist options
----------------

> ``hs_forcing_mod`` is used when ``idealised_moist_model`` in ``atmosphere_nml`` is set to ``FALSE``. 

Reviewer: ```suggestion
``hs_forcing_mod`` is used when ``idealized_moist_model`` in ``atmosphere_nml`` is set to ``FALSE``. 
```

:ka: Damping coefficient in the free atmosphere :math:`k_{a}`. If positive, :math:`k_{a}` has units :math:`\text{days}^{-1}`, and if negative, :math:`k_{a}` has units :math:`\text{days}` and is inverted. Default :math:`1/40\,\text{days}^{-1}`. 
:ks: Surface damping coefficient :math:`k_{s}`. If positive, :math:`k_{s}` has units :math:`\text{days}^{-1}`, and if negative, :math:`k_{s}` has units :math:`\text{days}` and is inverted. Default :math:`1/4\,\text{days}^{-1}`. 
:P00: Reference pressure for ``'Held_Suarez'`` and ``'EXOPLANET'`` equilibrium temperature fields. Default :math:`10^{5}\,\text{Pa}`. 
:p_trop: Reference (tropopause) pressure for ``'EXOPLANET2'`` equilibrium temperature field. Default :math:`10^{4}\,\text{Pa}`. 
:alpha: Pressure exponent for ``'EXOPLANET2'`` equilibrium temperature field. Default :math:`\alpha=2/7` yields a neutrally stratified atmosphere. 

**Top down** 

:sigma_b: Top-of-boundary layer pressure, as for the Rayleigh damping. The same namelist variable is used for each of the ``rayleigh_damping``, ``newtonian_damping`` and ``top_down_newtonian_damping`` subroutines. 

> :lapse: Lapse rate :math:`\Gamma`. Units are :math:`\text{K}\,\text{km}^{-1}`. Default :math:`6\,\text{K}\,text{km}^{-1}`. 

Reviewer: Missing `\` on final units
```suggestion
:lapse: Lapse rate :math:`\Gamma`. Units are :math:`\text{K}\,\text{km}^{-1}`. Default :math:`6\,\text{K}\,\text{km}^{-1}`. 
```

**Local heating**

:local_heating_option: String indicating the option used for local heating. Valid options are ``'from_file'`` and ``'Isidoro'`` (see documentation for meaning), otherwise local heating is not performed. Default ``''``. 

:local_heating_file: Name of input netCDF file for :math:`Q_{\text{local}}` wind used if ``local_heating_option = 'from_file'``. File name should exclude ``.nc`` suffix. 

The following options are used if ``local_heating_option = 'Isidoro'``

:local_heating_srfamp: Surface amplitude of local heating :math:`A`. Default :math:`0.0\,\text{K}\,\text{day}^{-1}`. 
:local_heating_xwidth: Longitudinal width of local heating :math:`\Delta\lambda`. Default :math:`10^{\circ}`. 
:local_heating_ywidth: Latitudinal width of local heating :math:`\Delta\theta`. Default :math:`10^{\circ}`. 
:local_heating_xcenter: Longitudinal center of heating :math:`\lambda_{0}`. Default :math:`180^{\circ}`. 
:local_heating_ycenter: Longitudinal center of heating :math:`\lambda_{0}`. Default :math:`45^{\circ}`. 
:local_heating_vert_decay: Vertical decay of local heating in pressure :math:`\Delta p`. Default :math:`10^{4}\,\text{Pa}`.  

**Other**

> There are also optins for a tracer source and sink, ``trflux`` and ``trsink``. These are currently undocumented, however, as Isca only uses the ``hs_forcing`` when the model is dry and there are no tracers. 

Reviewer: ```suggestion
There are also options for a tracer source and sink, ``trflux`` and ``trsink``. These are currently undocumented, however, as Isca only uses the ``hs_forcing`` when the model is dry and there are no tracers. 
```

> There are also optins for a tracer source and sink, ``trflux`` and ``trsink``. These are currently undocumented, however, as Isca only uses the ``hs_forcing`` when the model is dry and there are no tracers. 

Reviewer: Perhaps change `there are no tracers` to `there are no active tracers`, since it is possible to add passive tracers in the dry model.

+-------------------+--------------------------------------+------------------------------------+
| diss_heat         | Vertically integrated heating due to | :math:`\text{W}\,\text{m}^{-2}`    |
|                   | kinetic energy dissipation           |                                    |
+-------------------+--------------------------------------+------------------------------------+
| local_heating     | Local heating                        | :math:`\text{K}\,\text{s}^{-1}`    |
+-------------------+--------------------------------------+------------------------------------+
| newtonian_damping | Heating due to Newtonian damping     | :math:`\text{K}\,\text{s}^{-1}`    |
+-------------------+--------------------------------------+------------------------------------+
| h_trop            | Tropopause height in ``top_down``    | :math:`\text{W}\,\text{m}^{-2}`    |
|                   | model                                |                                    |
+-------------------+--------------------------------------+------------------------------------+

References
----------

| [Held1994]_ 
| [Penn2018]_
| [VallisEtAl2015]_
| [VallisEtAl2018]_

Authors
----------

> This documentation was written by Neil Lewis, peer reviewed by X, and quality controlled by Y. 

Reviewer: ```suggestion
This documentation was written by Neil Lewis, peer reviewed by Will Seviour, and quality controlled by Y. 
```
